### ISSUE TITLE
New FFU installation 

### ISSUE THREAD

**Issue by DxBONESxBACK** [Tobi823/ffupdater#384]
This morning I installed ffu_updater on my backup Android phone and then downloaded chromium through the ffu_updater after that it said it needed an update which I found a little strange that it wouldn't have initially downloaded the most recent version unless by chance it was updated 5 minutes after I downloaded it, anyway it failed to install twice.
Stacktrace:
```
java.lang.RuntimeException: Failed to install CHROMIUM in the foreground.
	at de.marmaro.krt.ffupdater.DownloadActivity.installApp(DownloadActivity.kt:405)
	at de.marmaro.krt.ffupdater.DownloadActivity.access$installApp(DownloadActivity.kt:59)
	at de.marmaro.krt.ffupdater.DownloadActivity$installApp$1.invokeSuspend(DownloadActivity.kt:0)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:104)
	at android.os.Handler.handleCallback(Handler.java:883)
	at android.os.Handler.dispatchMessage(Handler.java:100)
	at android.os.Looper.loop(Looper.java:214)
	at android.app.ActivityThread.main(ActivityThread.java:7631)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:964)
Caused by: de.marmaro.krt.ffupdater.installer.exceptions.InstallationFailedException: INSTALL_FAILED_VERSION_DOWNGRADE. The installation failed because one or more of the APKs was invalid.
	at de.marmaro.krt.ffupdater.installer.impl.SessionInstaller.fail(SessionInstaller.kt:187)
	at de.marmaro.krt.ffupdater.installer.impl.SessionInstaller.handleAppInstallationResult(SessionInstaller.kt:126)
	at de.marmaro.krt.ffupdater.installer.impl.SessionInstaller.access$handleAppInstallationResult(SessionInstaller.kt:29)
	at de.marmaro.krt.ffupdater.installer.impl.SessionInstaller$registerIntentReceiver$1.onReceive(SessionInstaller.kt:207)
	at android.app.LoadedApk$ReceiverDispatcher$Args.lambda$getRunnable$0$LoadedApk$ReceiverDispatcher$Args(LoadedApk.java:1590)
	at android.app.-$$Lambda$LoadedApk$ReceiverDispatcher$Args$_BumDX2UKsnxLVrE6UJsJZkotuA.run(Unknown Source:2)
	... 7 more
```
Device information:
| Key | Value |
| --- | --- |
| FFUpdater version | 77.7.10 (136) release |
| Device | LGL455DL (mdh10lm_fg, mdh10lm, mt6765) |
| Manufacturer | lge (LGE) | 
| Supported ABIs | armeabi-v7a, armeabi |
| Android version | 10 (SDK: 29) |
| OS | MCSBS9R18, jenkins, release-keys, 1675655242000 |

**Comment by DxBONESxBACK** [comment]
After I finished posting my error report and went back the ffu_updater now doesn't display an update available like before. 🤔 Hmmm

**Comment by Tobi823** [comment]
That's strange. Maybe it is #358?

**Comment by DxBONESxBACK** [comment]
IC2 Buggy Chromium VERS.


>After both my main and my backup Android phones, both running on the Red Velvet Cake platform had major issues shortly after I updated Chromium through FFUpdater earlier today. Both phones initially failed to install in the background and did fine manually (forefront) but shortly after another notification popped up saying Chromium failed to install in background and even finding this suspicious it successfully installed manually a second time and that's where it all began as webpages loaded very slowly and both phones needed rebooted. I'm pretty sure that the only similar thing that has happened to the two phones prior to them acting all buggy was the updates to Chromium. Even though I performed both of these updates about a half hour apart from each other they are also one update version different too!
![AllPopup_20230331_140254.jpg](https://user-images.githubusercontent.com/124132012/229240212-2d067870-a28e-4ac7-beb2-cbafa5a7368e.jpg)
_"Here (above) is a screenshot from my REVVL V+5G showing you the version # and (below) is a screenshot I took right after I took the one (above) this one is from my  backup phone (LG L455DL) both are running Android 11."_
![AllPopup_20230331_140146.jpg](https://user-images.githubusercontent.com/124132012/229240588-bdb6006a-a165-4d15-a84b-198cc154981a.jpg)
The two phones are not running on the same network so that's not the issue. It seems that these two updated versions have similar bugs that weren't there a couple days ago.
Not at my end anyway. 
So before wrapping this all up I also have a couple things to say...
1)My intentions here are not to complain about your work, it's merely to share with you what I am experiencing at my end. 
2)I'd like to add that earlier this morning the first error that I encountered on my main phone I posted a Issue #387. I thank you again for the work you do and all your help. 
P.S. I think it might be a good idea to forward this issue on to Chromiums developer/s as well.
Thanks Again 
D~Bone

Stacktrace:
```
java.lang.RuntimeException: Failed to install CHROMIUM in the foreground.
	at de.marmaro.krt.ffupdater.DownloadActivity.installApp(DownloadActivity.kt:405)
	at de.marmaro.krt.ffupdater.DownloadActivity.access$installApp(DownloadActivity.kt:59)
	at de.marmaro.krt.ffupdater.DownloadActivity$installApp$1.invokeSuspend(DownloadActivity.kt:0)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:104)
	at android.os.Handler.handleCallback(Handler.java:883)
	at android.os.Handler.dispatchMessage(Handler.java:100)
	at android.os.Looper.loop(Looper.java:214)
	at android.app.ActivityThread.main(ActivityThread.java:7631)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:964)
Caused by: de.marmaro.krt.ffupdater.installer.exceptions.InstallationFailedException: INSTALL_FAILED_VERSION_DOWNGRADE. The installation failed because one or more of the APKs was invalid.
	at de.marmaro.krt.ffupdater.installer.impl.SessionInstaller.fail(SessionInstaller.kt:187)
	at de.marmaro.krt.ffupdater.installer.impl.SessionInstaller.handleAppInstallationResult(SessionInstaller.kt:126)
	at de.marmaro.krt.ffupdater.installer.impl.SessionInstaller.access$handleAppInstallationResult(SessionInstaller.kt:29)
	at de.marmaro.krt.ffupdater.installer.impl.SessionInstaller$registerIntentReceiver$1.onReceive(SessionInstaller.kt:207)
	at android.app.LoadedApk$ReceiverDispatcher$Args.lambda$getRunnable$0$LoadedApk$ReceiverDispatcher$Args(LoadedApk.java:1590)
	at android.app.-$$Lambda$LoadedApk$ReceiverDispatcher$Args$_BumDX2UKsnxLVrE6UJsJZkotuA.run(Unknown Source:2)
	... 7 more
```
Device information:
| Key | Value |
| --- | --- |
| FFUpdater version | 77.7.10 (136) release |
| Device | LGL455DL (mdh10lm_fg, mdh10lm, mt6765) |
| Manufacturer | lge (LGE) | 
| Supported ABIs | armeabi-v7a, armeabi |
| Android version | 10 (SDK: 29) |
| OS | MCSBS9R18, jenkins, release-keys, 1675655242000 |

**Comment by Tobi823** [comment]
I will look later into it. Writing a master thesis and doing other stuff takes a lot of time :)

**Comment by Tobi823** [comment]
I fixed a bug in 78.0.1 that could cause data corruption if two downloads were running at the same time. I think this is the cause for your problem.

Please open a new ticket if the bug still persists.
